### PR TITLE
BGR-3859: streaming API for badgeclasses and issuers

### PR DESCRIPTION
**Pull request by thox**

### CODE REVIEW

File: apps/issuer/serializers_v2.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ class Meta(DetailSerializerV2.Meta):
         })
 
     def to_internal_value(self, data):
-        if 'expires' in data:
+        if not isinstance(data, BadgeClass) and 'expires' in data:
```
Comment (ottonomy) [inline apps/issuer/serializers_v2.py]: Not sure what this is protecting against, other than a potential misuse of the serializer initialization. The data passed into the serializer should be simple data types such as might be sent over a JSON API request, not fully realized python objects/django models.

Comment (louisconcentricsky) [reply apps/issuer/serializers_v2.py]: This test fails if that safety check is removed:  test_application_can_get_changed_badgeclasses (issuer.tests.test_badgeclass.BadgeClassesChangedApplicationTests) instigating further

Comment (thox) [reply apps/issuer/serializers_v2.py]: Yes, I remember having an issue during development with this serializer. When I debugged it, I noticed that it was not being sent a dict, but actually a BadgeClass instance, causing the new endpoint to crash.

Comment (louisconcentricsky) [reply apps/issuer/serializers_v2.py]: So, safety check does not sit well with me. I feel it is a band-aid over a potentially not good situation/use of the serializer. But in examining the code, it isn't obvious where that is. Thus, to unblock Charlie, I am going to approve and merge this in as we are not changing the API contract. Then, I will be looking to further investigate and allay our fears.

**Pull request merged**